src/js/date.js

```diff
@@ -95,3 +100,41 @@ export const getNumericWeekday = (locale = 'en-uk', date = new Date()) => {
   });
   return weekdayIndex;
 };
+
+export const getLocaleDayMonthYear = (locale = 'en-uk', date = new Date()) => {
+  const objDate = date.toLocaleString(locale, { day: 'numeric', month: 'numeric', year: 'numeric' });
+  return objDate;
+};
+
+export const isDateValid = (locale = 'en-uk', inputValue = '') => {
+  // year, monthIndex, day
+  const blueprint = new Date(2017, 2, 23);
+
+  if (!Intl.DateTimeFormat.supportedLocalesOf(locale).length) {
+    throw new Error(`locale not supported: ${locale}`);
+  }
+
+  // find out which out of 4 valid seperator the current locale has
+  const localisedBlueprintDate = new Intl.DateTimeFormat(locale).format(blueprint);
+  const localisedBlueprintDateString = localisedBlueprintDate.toString();
+
+  const usedSeperator = localisedBlueprintDateString.match(ALL_DATE_SEPERATORS);
```

Comment:
@LucaMele 
Be carefull `.match` returns an array of matches or `null` if nothing matched.
So you should pick `usedSeperator [0]` and check for `null`

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match

Reply:
@AndyOGo nice catch! thanks!!